packages/next-intl/src/shared/utils.tsx

```diff
@@ -90,7 +90,11 @@ export function prefixPathname(locale: string, pathname: string) {
 
 export function hasPathnamePrefixed(locale: string, pathname: string) {
   const prefix = `/${locale}`;
-  return pathname === prefix || pathname.startsWith(`${prefix}/`);
+  if (pathname === prefix) return true;
+  const hasPrefix = pathname.startsWith(prefix);
+  // check if the locale is the second segment of the URL to support basePath
+  if (!hasPrefix && pathname.split('/')?.[2] === locale) return true;
```

Comment:
The base path can contain multiple segments (e.g. `/base/path`), we need to be a bit more flexible here.

I found a way to detect the base path like this:

https://github.com/amannn/next-intl/blob/16b19e861dcec88fc6e436cadf99543340819bcd/packages/next-intl/src/navigation/shared/syncLocaleCookie.tsx#L28

Maybe that can be used when router functions are called?

Reply:
you are right, haven't thought about that. I added a similar version to the baseRouter.